tasks/sync-granule/tests/sync_granule_test.js

```diff
@@ -467,3 +470,64 @@ test.serial('attempt to download file from non-existent path - throw error', asy
 //     return t.end(e);
 //   });
 // });
+
+test.serial('when duplicateHandling is "version", keep both data if different', async (t) => {
+  // duplicateHandling is taken from task config or collection config
+  t.context.event.config.duplicateHandling = 'version';
+  const granuleFilePath = randomString();
+  const granuleFileName = payload.input.granules[0].files[0].name;
+
+  t.context.event.config.provider = {
+    id: 'MODAPS',
+    protocol: 's3',
+    host: randomString()
+  };
+
+  t.context.event.input.granules[0].files[0].path = granuleFilePath;
+
+  validateConfig(t, t.context.event.config);
+  validateInput(t, t.context.event.input);
```

Comment:
Should these be `await`ed? As they are `async` functions. Not sure since we only care about them throwing an error. Maybe `t.is(validateXY(...), true)`?